src/mixins/amalgamation-mixin.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, Vue } from 'vue-property-decorator'
+import { Getter } from 'pinia-class'
+import { useStore } from '@/store/store'
+import { AmalgamatingStatuses } from '@/enums'
+import { AmalgamatingBusinessIF } from '@/interfaces'
+import { CorpTypeCd } from '@bcrs-shared-components/corp-type-module'
+
+/**
+ * Mixin that provides amalgamation rules, etc.
+ */
+@Component({})
+export default class AmalgamationMixin extends Vue {
+  @Getter(useStore) getAmalgamatingBusinesses!: AmalgamatingBusinessIF[]
+  @Getter(useStore) isRoleStaff!: boolean
+  @Getter(useStore) isTypeBcCcc!: boolean
+  @Getter(useStore) isTypeBcUlcCompany!: boolean
+
+  // *** TODO: finish this, maybe name them something recognizable...
+  /** Iterable array of rule functions, sorted by importance. */
+  readonly rules = [
+    this.rule1,
+    this.rule2,
+    this.rule3,
+    this.rule4,
+    this.rule5,
+    this.rule6
+  ]
+
+  // *** I'M STILL WONDERING IF I WANT TO USE THESE
+  // readonly isLear = (item: AmalgamatingBusinessIF): boolean => (item?.type === 'lear')
+  // readonly isForeign = (item: AmalgamatingBusinessIF): boolean => (item?.type === 'foreign')
+
+  /** True if there a limited company in the table. */
+  get isAnyLimited (): boolean {
+    return this.getAmalgamatingBusinesses.some(business =>
+      (business.type === 'lear' && business.legalType === CorpTypeCd.BC_COMPANY)
+    )
+  }
+
+  /** True if there an unlimited company in the table in Alberta, Nova Scotia or USA. */
+  get isAnyUnlimitedAbNsUsa (): boolean {
+    return this.getAmalgamatingBusinesses.some(business =>
+      (business.type === 'lear' && business.legalType === CorpTypeCd.BC_COMPANY)
+    )
+  }
+
+  // readonly amalgamationRules = [
+  //   {
+  //     id: 0,
+  //     rule: (v: any) => !!v || 'Required',
+  //     status: AmalgamatingStatuses.ERROR_FOREIGN
+  //   }
+  // ]
```

Comment:
More WIP. Just assume everything in here is WIP, except for the basic architecture and design :)